client/wc-api/items/mutations.js

```diff
@@ -5,9 +5,14 @@
  */
 import { getResourceName } from '../utils';
 
-const updateItem = operations => ( type, id, itemData ) => {
+const updateItem = operations => async ( type, id, itemData, callback ) => {
```

Comment:
What do you think about using Promises here instead of a callback argument?

Reply:
I took a look at how Gutenberg deals with success/error messages for async actions and it looks like they add the notices directly from the action creator instead of the component (see [savePost](https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/store/actions.js#L258) for example). So I opted for moving all the logic from the component to the mutation in b19643ed72715a045d9114123789d222933a0497, I would like to hear your thoughts on how it looks.